climakitae/data_loaders.py

```diff
@@ -664,103 +677,40 @@ def _read_catalog_from_select(selections, cat):
+        if orig_var_id_selection == "wind_speed_derived":  # Hourly
+            da = _get_wind_speed_derived(selections, cat)
+        elif orig_var_id_selection == "wind_direction_derived":  # Hourly
+            da = _get_wind_dir_derived(selections, cat)
+        elif orig_var_id_selection == "dew_point_derived":  # Monthly/daily
+            da = _get_monthly_daily_dewpoint(selections, cat)
+        elif orig_var_id_selection == "dew_point_derived_hrly":  # Hourly
+            da = _get_hourly_dewpoint(selections, cat)
+        elif orig_var_id_selection == "rh_derived":  # Hourly
+            da = _get_hourly_rh(selections, cat)
+        elif orig_var_id_selection == "q2_derived":  # Hourly
+            da = _get_hourly_specific_humidity(selections, cat)
+        elif orig_var_id_selection == "fosberg_index_derived":  # Hourly
+            da = _get_fosberg_fire_index(selections, cat)
         else:
-            # Load temperature data
-            selections.variable_id = ["t2"]
-            selections.units = (
-                "K"  # Kelvin required for humidity and dew point computation
-            )
-            t2_da = _get_data_one_var(selections, cat)
-
-            # Load mixing ratio data
-            selections.variable_id = ["q2"]
-            selections.units = "kg kg-1"
-            q2_da = _get_data_one_var(selections, cat)
-
-            # Load pressure data
-            selections.variable_id = ["psfc"]
-            selections.units = "Pa"
-            pressure_da = _get_data_one_var(selections, cat)
-
-            # Derive relative humidity
-            # Returned in units of [0-100]
-            rh_da = _compute_relative_humidity(
-                pressure=pressure_da,  # Pa
-                temperature=t2_da,  # Kelvin
-                mixing_ratio=q2_da,  # kg/kg
+            raise ValueError(
+                "You've encountered a bug. No data available for selected derived variable."
             )
 
-            if orig_var_id_selection == "rh_derived":
-                da = rh_da
-
-            else:
-                # Derive dew point temperature
-                # Returned in units of Kelvin
-                dew_pnt_da = _compute_dewpointtemp(
-                    temperature=t2_da, rel_hum=rh_da  # Kelvin  # [0-100]
-                )
-
-                if orig_var_id_selection == "dew_point_derived_hrly":
-                    da = dew_pnt_da
-
-                elif orig_var_id_selection == "q2_derived":
-                    # Derive specific humidity
-                    # Returned in units of g/kg
-                    da = _compute_specific_humidity(
-                        tdps=dew_pnt_da, pressure=pressure_da  # Kelvin  # Pa
-                    )
-
-                else:
-                    raise ValueError(
-                        "You've encountered a bug. No data available for selected derived variable."
-                    )
-
         da = _convert_units(da, selected_units=orig_unit_selection)
```

Comment:
I'm not sure where native_units gets defined, but I think if the fire index goes into here with native_units as [0-100] then it might get scaled down to cap the max at 100, rather than reassigning all values >100 to 100?